db/data/20230718131711_import_ukri_level_d_budgets.rb

```diff
@@ -0,0 +1,85 @@
+# Run me with `rails runner db/data/20230718131711_import_ukri_level_d_budgets.rb`
+
+# Script to add budgets to activities when those budgets cannot be added via bulk upload (for example they are for a
+# level not currently handled by bulk upload).
+
+# Re-usable if you replace `file_name` with the path to your chosen csv.
+# Note that the script expects column headers of the following form: "Activity RODA ID"; "Budget amount"; "Type";
+# "Financial year" The order of the columns is not sensitive.
+
+# This script bypasses the usual budget validation required by the Budget model. This is to allow us to upload these
+# budgets without the need for a report attached to each.
+
+# Details of budgets that cannot be successfully created are logged to the terminal at the end of the script run.
+
+require "csv"
+
+def get_financial_year(financial_year_range)
+  financial_year_range.split("-")[0]
+end
+
+def validate_budget(budget)
+  budget.valid?
+  report_error = budget.errors.messages[:report]
+  budget.errors.messages.count == 1 && report_error.count == 1
```

Comment:
This works, but you can also simplify `budget.errors.messages.count` to `budget.errors.count`